File: app/src/main/java/com/mktplace/messenger/ui/FacebookWebChromeClient.java
package com.mktplace.messenger.ui;

import android.webkit.WebChromeClient;
import android.webkit.WebView;

public class FacebookWebChromeClient extends WebChromeClient {

    public interface ProgressListener {
        void onProgress(int progress);
    }

    private final ProgressListener listener;

    public FacebookWebChromeClient(ProgressListener listener) {
        this.listener = listener;
    }

    @Override
    public void onProgressChanged(WebView view, int newProgress) {
        super.onProgressChanged(view, newProgress);
        listener.onProgress(newProgress);
    }
}

File: app/src/main/java/com/mktplace/messenger/ui/FacebookWebChromeClient.kt
package com.mktplace.messenger.ui

import android.webkit.WebChromeClient
import android.webkit.WebView

class FacebookWebChromeClient(
    private val onProgressChanged: (Int) -> Unit
) : WebChromeClient() {

    override fun onProgressChanged(view: WebView, newProgress: Int) {
        super.onProgressChanged(view, newProgress)
        onProgressChanged(newProgress)
    }
}
